es/bulk_maker.go

```diff
@@ -0,0 +1,127 @@
+package es
+
+import (
+	"bytes"
+	"time"
+
+	"github.com/astroband/astrologer/db"
+	"github.com/stellar/go/xdr"
+)
+
+var (
+	fakeOperationIndex      uint8 = 255
+	balanceFromMetaAuxOrder uint8 = 1
+	balanceFromFeeAuxOrder  uint8 = 2
+)
+
+// BulkMaker creates es bulk from ledger data
+type BulkMaker struct {
+	ledgerRow       db.LedgerHeaderRow
+	ledgerHeader    *LedgerHeader
+	seq             int
+	closeTime       time.Time
+	transactionRows []db.TxHistoryRow
+	transactions    []*Transaction
+	fees            []db.TxFeeHistoryRow
+	buffer          *bytes.Buffer
+}
+
+// NewBulkMaker returns new BulkMaker structure
+func NewBulkMaker(l db.LedgerHeaderRow, t []db.TxHistoryRow, f []db.TxFeeHistoryRow, b *bytes.Buffer) *BulkMaker {
+	h := NewLedgerHeader(&l)
+
+	txs := make([]*Transaction, len(t))
+	for i := 0; i < len(t); i++ {
+		txs[i] = NewTransaction(&t[i], h.CloseTime)
+	}
+
+	return &BulkMaker{
+		ledgerRow:       l,
+		ledgerHeader:    h,
+		seq:             h.Seq,
+		closeTime:       h.CloseTime,
+		transactionRows: t,
+		transactions:    txs,
+		fees:            f,
+		buffer:          b,
+	}
+}
+
+// Make creates bulk
+func (m *BulkMaker) Make() {
+	m.makeLedger()
+	m.makeTransactions()
+	m.makeOperationsWithResults()
+	m.makeBalancesFromMetas()
+	m.makeBalancesFromFeeHistory()
+}
+
+func (m *BulkMaker) makeLedger() {
+	SerializeForBulk(m.ledgerHeader, m.buffer)
+}
+
+func (m *BulkMaker) makeTransactions() {
+	for _, transaction := range m.transactions {
+		SerializeForBulk(transaction, m.buffer)
+	}
+}
+
+func (m *BulkMaker) makeOperationsWithResults() {
+	for tIndex, t := range m.transactions {
+		row := m.transactionRows[tIndex]
+		operations := row.Envelope.Tx.Operations
+		results := row.Result.Result.Result.Results
```

Comment:
Wow, that's a lot of results 😅 Just noticing, I understand that it's not under our control